src/Adapter/FlysystemItems.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\ContentApiBundle\Adapter;
+
+use IteratorAggregate;
+use League\Flysystem\FileNotFoundException;
+use League\Flysystem\FilesystemInterface;
+use Libero\ContentApiBundle\Exception\ItemNotFound;
+use Libero\ContentApiBundle\Exception\UnexpectedVersionNumber;
+use Libero\ContentApiBundle\Exception\VersionNotFound;
+use Libero\ContentApiBundle\Model\ItemId;
+use Libero\ContentApiBundle\Model\ItemListPage;
+use Libero\ContentApiBundle\Model\Items;
+use Libero\ContentApiBundle\Model\ItemVersion;
+use Libero\ContentApiBundle\Model\ItemVersionNumber;
+use RuntimeException;
+use Traversable;
+use function array_filter;
+use function array_map;
+use function array_pop;
+use function array_search;
+use function array_slice;
+use function count;
+use function iterator_to_array;
+use function Libero\ContentApiBundle\hash_for_stream;
+use function strnatcmp;
+use function time;
+use function usort;
+
+final class FlysystemItems implements IteratorAggregate, Items
```

Comment:
it's really hard to say anything about this without knowing its intended scope (which should then be documented).

For example:

- `list()` seems to request a whole listing of entries to generate each page; this may be fast enough locally, but it can easily take several seconds on a large folder, or a medium sized S3 bucket
- depending on the adapter being used, consistency like [read-after-write might not be guaranteed](https://docs.aws.amazon.com/AmazonS3/latest/dev/Introduction.html#ConsistencyModel) hence you could PUT a document and not be able to GET it, or DELETE it and still see it for a period of time; this seems to violate the contract if this article store is providing strict consistency like a relational database implementation does
- more exotic options like FTP or WebDAV has even more failure modes to take into account

So what is the intended usage and set of supported adapters?

Reply:
The target for this adapter would be a dummy implementation of the API where the content it static (the adapter is writeable, but I'd anticipate it being read-only at the API level).

I wouldn't expect remote content to be used, I'd be ok with removing the extra layer and just having a `FilesystemAdapter` instead (which can use iterators and locks).

Reply:
Similar caveats will exist with a DBAL implementation: not all drivers (eg SQLite) will return a resource (returning a string instead), so there's a potential for memory problems on _huge_ content. But that just needs documentation.

Reply:
We can lock down the `FlysystemAdapter` to only accept a few adapters that target the local filesystem. But if that's the case I agree it seems simpler to target directly a real filesystem with `fopen` and similar, without the layer of abstraction.